lib/Listener/WhenViewDeletedAuditLogListener.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Tables\Listener;
+
+use OCA\Tables\Event\ViewDeletedEvent;
+use OCA\Tables\Service\Support\AuditLogServiceInterface;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+/**
+ * @template-implements IEventListener<Event|ViewDeletedEvent>
+ */
+final class WhenViewDeletedAuditLogListener implements IEventListener {
+	public function __construct(protected AuditLogServiceInterface $auditLogService) {
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof ViewDeletedEvent)) {
+			return;
+		}
+
+		$view = $event->getView();
+		$userId = $event->getUserId();
+
+		$this->auditLogService->log("View with ID: $view->id was deleted by user with ID: $userId", [
+			'view' => $view->jsonSerialize(),
+			'userId' => $userId,
```

Comment:
As commented, we can probably skip this. The userId should be already in place in the log entry for any message (https://github.com/nextcloud/server/blob/d753d0476053fc7859e60fe97670a4299dd8fa32/lib/private/Log/LogDetails.php#L74)

Reply:
Thank you and Totally agreed with this duplication after reading the Log Service in details

Reply:
Thanks for the quick adjustment. Looks good now 👍